### DISCUSSION TITLE
How do I connect from my DDEV local setup?

### DISCUSSION THREAD

**Discussion by palomamtnez** [planetscale/discussion#410] [FAQs]
In my local environment, I have my app running through DDEV, and I have tried to connect to planetscale via pscale cli but couldn't find a proper guide for it so I couldn't make it work.

**Comment by fooforge** [comment]
Hey @palomamtnez,

I'm not all too familiar with DDEV, but if you want to develop fully locally without any ties to your database on PlanetScale, you would need to stand up a local MySQL Community server > 8.0.23. If you do that, I would recommend reading through our [MySQL<->Vitess compatibility guide](https://planetscale.com/docs/reference/mysql-compatibility) first.

If you want your DDEV environment to use your development branch of your PlanetScale database, you would likely need to provision an authentication token in your `.env` file if that concept exists in the DDEV world. You could then have our [CLI utility](https://planetscale.com/cli) `pscale` [act as a proxy](https://planetscale.com/docs/reference/connect) via `pscale connect [database] [branch]` which would, in turn, allow you to connect to your PlanetScale database via `localhost:3306` (or a random port if 3306's already occupied) as if it were available locally.

Let me know if that didn't help or if you're running into trouble getting this set up, and I'll see what else I can come up with.
